servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/CoreMessageReceiver.java

```diff
@@ -749,17 +749,8 @@ public void onReceiveComplete(Delivery delivery)
                                     matchingUpdateStateWorkItem.setLastKnownException(exception);
                                     // Retry after retry interval
                                     TRACE_LOGGER.debug("Pending updateState operation for delivery '{}' will be retried after '{}'", deliveryTagAsString, retryInterval);
-                                    try
-                                    {
-                                        this.underlyingFactory.scheduleOnReactorThread((int) retryInterval.toMillis(),
-                                                new DispatchHandler()
-                                                {
-                                                    @Override
-                                                    public void onEvent()
-                                                    {
-                                                        delivery.disposition(matchingUpdateStateWorkItem.getDeliveryState());
-                                                    }
-                                                });
+                                    try {
+                                        this.underlyingFactory.scheduleOnReactorThread((int) retryInterval.toMillis(), new DeliveryStateDispatchHandler(delivery, matchingUpdateStateWorkItem.getDeliveryState()));
```

Comment:
[SpotBugs-P2]
This class is an inner class, but does not use its embedded reference to the object which created it.  This reference makes the instances of the class larger, and may keep the reference to the creator object alive longer than necessary.  If possible, the class should be made static.